packages/mask/src/components/shared/SelectRecipients/SelectRecipients.tsx

```diff
@@ -1,62 +1,78 @@
-import { Box, Chip } from '@mui/material'
-import { makeStyles } from '@masknet/theme'
-import AddIcon from '@mui/icons-material/Add'
 import { useEffect, useState } from 'react'
-import { difference } from 'lodash-unified'
-import { useI18N } from '../../../utils'
-import type { ProfileInformation as Profile } from '@masknet/shared-base'
+import { ProfileInformation as Profile, EMPTY_LIST, NextIDPlatform } from '@masknet/shared-base'
 import type { LazyRecipients } from '../../CompositionDialog/CompositionUI'
 import { SelectRecipientsDialogUI } from './SelectRecipientsDialog'
 import { useCurrentIdentity } from '../../DataSource/useActivatedUI'
-import { EMPTY_LIST } from '@masknet/shared-base'
-
-const useStyles = makeStyles()({
-    root: {
-        display: 'inline-flex',
-        flexWrap: 'wrap',
-    },
-})
+import { useNextIDBoundByPlatform } from '../../DataSource/useNextID'
+import { useTwitterIdByWalletSearch } from './useTwitterIdByWalletSearch'
+import { isValidAddress } from '@masknet/web3-shared-evm'
+import { useI18N } from '../../../utils'
+import { cloneDeep } from 'lodash-unified'
 
 export interface SelectRecipientsUIProps {
     items: LazyRecipients
     selected: Profile[]
     disabled?: boolean
     hideSelectAll?: boolean
     hideSelectNone?: boolean
+    open: boolean
+    onClose(): void
     onSetSelected(selected: Profile[]): void
 }
+const resolveNextIDPlatform = (value: string) => {
+    if (isValidAddress(value)) return NextIDPlatform.Ethereum
+    if (value.length >= 44) return NextIDPlatform.NextID
+    if (/^\w{1,15}$/.test(value)) return NextIDPlatform.Twitter
+    return undefined
+}
 
 export function SelectRecipientsUI(props: SelectRecipientsUIProps) {
+    const { items, selected, onSetSelected, open, onClose } = props
     const { t } = useI18N()
-    const { classes } = useStyles()
-    const { items, selected, onSetSelected } = props
+    const [valueToSearch, setValueToSearch] = useState('')
     const currentIdentity = useCurrentIdentity()
+    const type = resolveNextIDPlatform(valueToSearch)
+    const { loading: searchLoading, value: NextIDResults } = useNextIDBoundByPlatform(
+        type ?? NextIDPlatform.NextID,
+        valueToSearch.toLowerCase(),
+    )
+    const NextIDItems = useTwitterIdByWalletSearch(NextIDResults, valueToSearch, type)
     const profileItems = items.recipients?.filter((x) => x.identifier !== currentIdentity?.identifier)
-    const [open, setOpen] = useState(false)
 
+    const unique = (arr: Profile[]) => {
+        const res = new Map()
+        return arr.filter(
+            (item) => !res.has(item.linkedPersona?.publicKeyAsHex) && res.set(item.linkedPersona?.publicKeyAsHex, 1),
+        )
+    }
+    const searchedList = unique(profileItems?.concat(NextIDItems) ?? [])
```

Comment:
https://lodash.com/docs/4.17.15#uniqBy

```suggestion
    const searchedList = uniqBy(profileItems?.concat(NextIDItems) ?? [], ({ linkedPersona }) => linkedPersona?.publicKeyAsHex)
```

Reply:
done